cpp/ql/src/semmle/code/cpp/ir/implementation/internal/IRFunctionBase.qll

```diff
@@ -0,0 +1,27 @@
+/**
+ * Provides a base class, `IRFunctionBase`, for the stage-independent portions of `IRFunction`.
+ */
+
+private import IRFunctionBaseInternal
+
+private newtype TIRFunction =
+  MkIRFunction(Language::Function func) { IRConstruction::Raw::functionHasIR(func) }
```

Comment:
I'm surprised this isn't `cached`. But it wasn't cached before, so this PR is not the right time to experiment with its caching.